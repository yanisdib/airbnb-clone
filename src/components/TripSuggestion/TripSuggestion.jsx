import styled from 'styled-components'
import {v4 as uuidv4} from 'uuid';

import { Container, CardGrid } from '..';

import barcelonetteThumbnail from '../../assets/images/barcelonette-thumbnail.jpg';
import varsThumbnail from '../../assets/images/vars-thumbnail.jpg';
import allosThumbnail from '../../assets/images/allos-thumbnail.jpg';
import portoVecchioThumbnail from '../../assets/images/porto-vecchio-thumbnail.jpg';


const Section = styled.div`
    --mls-ct-bg: #D93B30;

    --tl-color: #FFFFFF;
    --tl-font-size: 35px;
    --tl-font-weight: 600;
    --tl-letter-spacing: -0.01em;
    --tl-line-height: 39px;

    --sub-font-size: 21px;
    --sub-font-weight: 300;
    --sub-letter-spacing: -0.01em;
    --sub-line-height: 22px;

    --mls-sc-spt-xs: 48px;
    --mls-sc-spb-xs: 0px;
    --mls-sc-spt-sm: 48px;
    --mls-sc-spb-sm: 0px;
    --mls-sc-spt-md: 64px;
    --mls-sc-spb-md: 0px;
    --mls-sc-spt-lg: 96px;
    --mls-sc-spb-lg: 0px;
    --mls-sc-spt-xl: 96px;
    --mls-sc-spb-xl: 0px;

    @media (min-width: 375px) {
        display: flex;
        align-items: center;
        justify-content: center;
        background: #FFFFFF;
        padding-top: var(--mls-sc-spt-xs);
        padding-bottom: var(--mls-sc-spb-xs);
    }

    @media (min-width: 744px) {
        display: flex;
        background: #FFFFFF;
        padding-top: var(--mls-sc-spt-sm);
        padding-bottom: var(--mls-sc-spb-sm);
    }

    @media (min-width: 950px) {
        display: flex;
        background: #FFFFFF;
        padding-top: var(--mls-sc-spt-md);
        padding-bottom: var(--mls-sc-spb-md);
    }

    @media (min-width: 1128px) {
        display: flex;
        background: #FFFFFF;
        padding-top: var(--mls-sc-spt-lg);
        padding-bottom: var(--mls-sc-spb-lg);
    }

    @media (min-width: 1440px) {
        display: flex;
        background: #FFFFFF;
        padding-top: var(--mls-sc-spt-xl);
        padding-bottom: var(--mls-sc-spb-xl);
    }
`;

const Wrapper = styled.div`
    display: flex;
`;

const Content = styled.div`
    display: flex;
`;

const CityCard = styled.div`
    width: 100%;
    height: 510px;
    background-color: var(--mls-ct-bg, none);
    border-radius: 12px;

    img {
        width: 100%;
        border-top-left-radius: 12px;
        border-top-right-radius: 12px;
        object-fit: contain;
    }
`;

const CardTitle = styled.h2`
    max-width: 100%;
    font-size: var(--tl-font-size, 32px);
    font-weight: var(--tl-font-weight, 800);
    color: var(--tl-color, #222222);
    letter-spacing: var(--tl-letter-spacing, 0em);
    line-height: var(--tl-line-height, 36px);
    white-space: pre-line;
    padding: 24px 16px 16px;
`;

const CityDistance = styled.span`
    font-size: var(--sub-font-size, 14px);
    font-weight: var(--sub-font-weight, 400);
    color: var(--tl-color, #717171);
    letter-spacing: var(--sub-letter-spacing, 0em);
    line-height: var(--sub-line-height, 18px);
    padding: 0px 16px 16px;
`;


function TripSuggestion() {
    const suggestedCities = [
        { 'Barcelonette': { distanceInKm: 152, img: barcelonetteThumbnail } },
        { 'Vars': { distanceInKm: 172, img: varsThumbnail } },
        { 'Allos': { distanceInKm: 136, img: allosThumbnail } },
        { 'Porto Vecchio': { distanceInKm: 323, img: portoVecchioThumbnail } }
    ];

    const suggestedCitiesCards = suggestedCities.map(city => {
        return (
            <CityCard key={uuidv4()}>
                <img src={city[`${Object.keys(city)}`]["img"]} />
                <CardTitle>{Object.keys(city)}</CardTitle>
                <CityDistance>
                    {city[`${Object.keys(city)}`]["distanceInKm"]} kilometers away
                </CityDistance>
            </CityCard>
        );
    });

    
    return (
        <Section>
            <Wrapper>
                <Container>
                    <Content>
                        <CardGrid>
                            {suggestedCitiesCards}
                        </CardGrid>
                    </Content>
                </Container>
            </Wrapper>
        </Section>
    )
}


export default TripSuggestion;