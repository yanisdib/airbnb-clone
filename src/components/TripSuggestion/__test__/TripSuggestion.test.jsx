import { mount } from 'enzyme';

import TripSuggestion from '../TripSuggestion';


describe('TripSuggestion component', () => {
    let wrapper;

    beforeEach(() => wrapper = mount(<TripSuggestion />))

    it('renders correctly without crashing', () => expect(wrapper).toMatchSnapshot())
}) 