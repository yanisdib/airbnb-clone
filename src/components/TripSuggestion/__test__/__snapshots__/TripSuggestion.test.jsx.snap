// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripSuggestion component renders correctly without crashing 1`] = `
<TripSuggestion>
  <styled.div>
    <div
      className="sc-iUKqMP zfAfw"
    >
      <styled.div>
        <div
          className="sc-iAKWXU eSxgFC"
        >
          <styled.div>
            <div
              className="sc-bdvvtL fNgVow"
            >
              <styled.div>
                <div
                  className="sc-efQSVx inTfYM"
                >
                  <styled.div>
                    <div
                      className="sc-hKwDye bCtFQq"
                    >
                      <styled.div
                        key="db2db1d9-a0ad-47e2-bf1d-8583ceda2453"
                      >
                        <div
                          className="sc-cTAqQK oIbSG"
                        >
                          <img
                            src="barcelonette-thumbnail.jpg"
                          />
                          <styled.h2>
                            <h2
                              className="sc-jObWnj HGpB"
                            >
                              Barcelonette
                            </h2>
                          </styled.h2>
                          <styled.span>
                            <span
                              className="sc-dPiLbb fMMWqx"
                            >
                              152
                               kilometers away
                            </span>
                          </styled.span>
                        </div>
                      </styled.div>
                      <styled.div
                        key="4d40bb12-d5bd-4440-a017-f7758f361c3f"
                      >
                        <div
                          className="sc-cTAqQK oIbSG"
                        >
                          <img
                            src="vars-thumbnail.jpg"
                          />
                          <styled.h2>
                            <h2
                              className="sc-jObWnj HGpB"
                            >
                              Vars
                            </h2>
                          </styled.h2>
                          <styled.span>
                            <span
                              className="sc-dPiLbb fMMWqx"
                            >
                              172
                               kilometers away
                            </span>
                          </styled.span>
                        </div>
                      </styled.div>
                      <styled.div
                        key="b0e4dff3-972d-42f0-9b3b-1e7befc5e66d"
                      >
                        <div
                          className="sc-cTAqQK oIbSG"
                        >
                          <img
                            src="allos-thumbnail.jpg"
                          />
                          <styled.h2>
                            <h2
                              className="sc-jObWnj HGpB"
                            >
                              Allos
                            </h2>
                          </styled.h2>
                          <styled.span>
                            <span
                              className="sc-dPiLbb fMMWqx"
                            >
                              136
                               kilometers away
                            </span>
                          </styled.span>
                        </div>
                      </styled.div>
                      <styled.div
                        key="a2cfcf82-429e-4f69-8b7b-1dc72581aef8"
                      >
                        <div
                          className="sc-cTAqQK oIbSG"
                        >
                          <img
                            src="porto-vecchio-thumbnail.jpg"
                          />
                          <styled.h2>
                            <h2
                              className="sc-jObWnj HGpB"
                            >
                              Porto Vecchio
                            </h2>
                          </styled.h2>
                          <styled.span>
                            <span
                              className="sc-dPiLbb fMMWqx"
                            >
                              323
                               kilometers away
                            </span>
                          </styled.span>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</TripSuggestion>
`;
